manage_breast_screening/notifications/tests/management/commands/test_send_message_batch.py

```diff
@@ -0,0 +1,115 @@
+import uuid
+from datetime import datetime, timedelta
+from unittest.mock import ANY, MagicMock, patch
+
+import pytest
+import requests
+
+from manage_breast_screening.notifications.api_client import ApiClient
+from manage_breast_screening.notifications.management.commands.send_message_batch import (
+    TZ_INFO,
+    Command,
+    CommandError,
+)
+from manage_breast_screening.notifications.models import Message, MessageBatch
+from manage_breast_screening.notifications.tests.factories import AppointmentFactory
+
+
+@patch.object(
+    ApiClient, "send_message_batch", return_value=MagicMock(spec=requests.Response)
+)
+class TestSendMessageBatch:
+    @pytest.fixture
+    def routing_plan_id(self):
+        return str(uuid.uuid4())
+
+    @pytest.mark.django_db
+    def test_handle_with_a_batch_to_send(
+        self, mock_send_message_batch, routing_plan_id
+    ):
+        """Test sending message batch with valid Appointment data"""
+        mock_send_message_batch.return_value.status_code = 201
+
+        appointment = AppointmentFactory(
+            starts_at=datetime.today().replace(tzinfo=TZ_INFO)
+        )
+
+        subject = Command()
+        mock_mark_batch_as_sent = MagicMock()
+        subject.mark_batch_as_sent = mock_mark_batch_as_sent
+
+        subject.handle(**{"routing_plan_id": routing_plan_id})
+
+        message_batches = MessageBatch.objects.filter(routing_plan_id=routing_plan_id)
+        assert message_batches.count() == 1
+        assert (
+            Message.objects.filter(
+                appointment=appointment, batch=message_batches[0]
+            ).count()
+            == 1
+        )
+        mock_mark_batch_as_sent.assert_called_once_with(message_batches[0], ANY)
+
+    @pytest.mark.django_db
+    def test_handle_with_nothing_to_send(
+        self, mock_send_message_batch, routing_plan_id
+    ):
+        """Test that no MessageBatch or Message records are created when no appointments need notifications"""
+        Command().handle(**{"routing_plan_id": routing_plan_id})
+
+        assert MessageBatch.objects.count() == 0
+        assert Message.objects.count() == 0
+
+    @pytest.mark.django_db
+    def test_handle_with_appointments_inside_schedule_window(
+        self, mock_send_message_batch, routing_plan_id
+    ):
+        """Test that appointments with date inside the schedule period are notified"""
+        mock_send_message_batch.return_value.status_code = 201
+        appointment = AppointmentFactory(
```

Comment:
minor question, is there some coverage that could be added here by creating appointments that are NOT in the window to ensure they don't get included?

Reply:
Yes good catch, the nothing to send scenario isn't quite the same thing as something to send in the future, I'll add a scenario with an appointment beyond the 4 week and 4 day window.

Reply:
Added a suitable test here: https://github.com/NHSDigital/dtos-manage-breast-screening/pull/198/commits/604b160547720b8a05114cb396d9b13dc517bcb3